src/btree/bt_curnext.c

```diff
@@ -178,6 +178,7 @@ __cursor_var_next(WT_CURSOR_BTREE *cbt, bool newpage)
 
 	/* Initialize for each new page. */
 	if (newpage) {
+		cbt->slot = UINT32_MAX;
```

Comment:
I'd like a comment around each of these, along the lines of `/* Be paranoid and set the slot out of bounds when moving to a new page. */`